src/psd_tools/api/layers.py

```diff
@@ -664,8 +792,15 @@ def _get_bbox(layer, **kwargs):
         lefts, tops, rights, bottoms = zip(*bboxes)
         return (min(lefts), min(tops), max(rights), max(bottoms))
 
-    def __init__(self, *args):
-        super(Group, self).__init__(*args)
+    def __init__(self, psd, record, channels, parent, _bounding_record = None, _bounding_channels = None):
```

Comment:
`_bounding_record` and `_bounding_channels` do not look friendly to the API users. Although instantiating a `Group` object is not likely to be a common case, I would suggest splitting the setter method for those private data.

Reply:
I added the new() method for creating a new group, as the __init__ needs several types of data to be already created (records and channels for example). new() creates all those necessary datas and the two attributes you mentionned.

The only other instance of direct group instantiation is in PSDImage._init for which these data already exist.

Though now that i think about it, the only place where this is absolutely needed is in the _update_record, and we can compute them in this function. I need to check if there's any important data in these record and channels of the folder.

Reply:
there are layer ids and other metadata that it's probably best to keep around for compatibility and data consistency

Reply:
Okay, if there are still cases for keeping those records, drop the arguments from the `__init__` and add a private setter method for them

Reply:
Removed them from the constructor and moved it to _set_bounding_records